File: test/corpus/c/duplicates.c
// An exact clone pair: identical up to renamed identifiers, above the clone size
// floor.

// dendro-expect: duplicate
int total_price(const int *items, int n, int tax) {
    int subtotal = 0;
    for (int i = 0; i < n; i++) {
        subtotal = subtotal + items[i];
    }
    int grand = subtotal + subtotal * tax;
    int rounded = round_value(grand);
    return rounded;
}

// dendro-expect: duplicate
int final_amount(const int *rows, int n, int rate) {
    int accum = 0;
    for (int i = 0; i < n; i++) {
        accum = accum + rows[i];
    }
    int whole = accum + accum * rate;
    int snapped = round_value(whole);
    return snapped;
}

File: test/corpus/php/duplicates.php
<?php

// An exact clone pair: identical up to renamed identifiers, above the clone size
// floor.

// dendro-expect: duplicate
function total_price($items, $tax) {
    $subtotal = 0;
    for ($i = 0; $i < count($items); $i++) {
        $subtotal = $subtotal + $items[$i];
    }
    $grand = $subtotal + $subtotal * $tax;
    $rounded = round_value($grand);
    return $rounded;
}

// dendro-expect: duplicate
function final_amount($rows, $rate) {
    $accum = 0;
    for ($i = 0; $i < count($rows); $i++) {
        $accum = $accum + $rows[$i];
    }
    $whole = $accum + $accum * $rate;
    $snapped = round_value($whole);
    return $snapped;
}
